interactive_templates/templates/v2/analysis/top_5.py

```diff
@@ -82,9 +82,9 @@ def create_top_5_code_table(
         A table of the top `nrows` codes.
     """
 
-    # cast both code columns to str
-    df[code_column] = df[code_column].astype(int).astype(str)
-    code_df[code_column] = code_df[code_column].astype(int).astype(str)
+    # prevent scientific notation for codes by casting to float then int
```

Comment:
Codelists with 17/18 codes are not that common, but I think you will run into the same thing we did here with codes being truncated. It happens as soon as the cohort-joiner step, so will the codes would already be truncated before reaching here. 

The strepa/scarlet solution was to make a local copy of the cohort joiner and [cast everything to strings ](https://github.com/opensafely/strepa_scarlet/blob/main/analysis/cohort_joiner.py#L12), which is not ideal from a memory usage perspective.

If you will always know the column name of the code column, you could provide it in a dictionary ([see docs](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html))
Or you could use feather files

Reply:
I noticed in some of the outputs the code descriptions aren't being matched for all codes. I forgot with had the same issue in the strepa project, perhaps that's the issue here. I'll look at updating it